src/VariableBorrowInterestRateStrategy.sol

```diff
@@ -125,46 +147,66 @@ contract VariableBorrowInterestRateStrategy is IDefaultInterestRateStrategy {
+    )
+        external view override
+        returns (uint256 liquidityRate, uint256 stableBorrowRate, uint256 variableBorrowRate)
+    {
         CalcInterestRatesLocalVars memory vars;
 
+        // 1. Calculate the total debt and set the current variable borrow rate to the base rate.
+
         vars.totalDebt = params.totalStableDebt + params.totalVariableDebt;
 
         vars.currentLiquidityRate      = 0;
         vars.currentVariableBorrowRate = _getBaseVariableBorrowRate();
 
+        // 2. Calculate the borrow and supply usage ratios.
+
         if (vars.totalDebt != 0) {
+            // Calculate the total resulting cash in the reserve after the call is complete.
             vars.availableLiquidity =
                 IERC20(params.reserve).balanceOf(params.aToken) +
                 params.liquidityAdded -
                 params.liquidityTaken;
 
+            // Calculate the total value of the asset in the reserve (cash + debt).
             vars.availableLiquidityPlusDebt = vars.availableLiquidity + vars.totalDebt;
+
+            // Calculate the borrowUsageRatio (debt / total value).
             vars.borrowUsageRatio = vars.totalDebt.rayDiv(vars.availableLiquidityPlusDebt);
-            vars.supplyUsageRatio = vars.totalDebt.rayDiv(
-                vars.availableLiquidityPlusDebt + params.unbacked
-            );
+
+            // Calculate the supplyUsageRatio (debt / total value + unbacked).
+            // NOTE: The supplyUsageRatio will almost always equal the borrowUsageRatio, except
+            //       when unbacked aTokens are minted.
+            vars.supplyUsageRatio
+                = vars.totalDebt.rayDiv(vars.availableLiquidityPlusDebt + params.unbacked);
         }
 
+        // 3. Calculate the variable borrow rate, using the 2-slope model.
+
         if (vars.borrowUsageRatio > OPTIMAL_USAGE_RATIO) {
-            uint256 excessBorrowUsageRatio = (vars.borrowUsageRatio - OPTIMAL_USAGE_RATIO).rayDiv(
-                MAX_EXCESS_USAGE_RATIO
-            );
+            // excessBorrowUsageRatio = (borrowRatio - optimalRatio) / (1 - optimalRatio)
+            uint256 excessBorrowUsageRatio
+                = (vars.borrowUsageRatio - OPTIMAL_USAGE_RATIO).rayDiv(MAX_EXCESS_USAGE_RATIO);
 
             vars.currentVariableBorrowRate +=
                 _getVariableRateSlope1() +
                 _getVariableRateSlope2().rayMul(excessBorrowUsageRatio);
         } else {
-            vars.currentVariableBorrowRate += _getVariableRateSlope1().rayMul(vars.borrowUsageRatio).rayDiv(
-                OPTIMAL_USAGE_RATIO
-            );
+            vars.currentVariableBorrowRate +=
+                _getVariableRateSlope1().rayMul(vars.borrowUsageRatio).rayDiv(OPTIMAL_USAGE_RATIO);
         }
 
+        // 4. Calculate the liquidity rate by multiplying the current variable borrow rate by
+        //    the supply usage ratio and subtracting the protocol's reserve factor. This yields the
```

Comment:
The reserve factor is not extracted via subtract, but by multiplication.

Reply:
Fixed